lib/internal/Magento/Framework/HTTP/Client/Curl.php

```diff
@@ -239,13 +249,108 @@ public function get($uri)
      * @param array|string $params
      * @return void
      *
-     * @see \Magento\Framework\HTTP\Client#post($uri, $params)
+     * @see \Magento\Framework\HTTP\Client::post($uri, $params)
+     * @url https://www.rfc-editor.org/rfc/rfc9110.html#section-9.3.3
      */
     public function post($uri, $params)
     {
         $this->makeRequest("POST", $uri, $params);
     }
 
+    /**
+     * Make PUT request
+     *
+     * @param string $uri
+     * @param array|string $params
+     * @return void
+     *
+     * @url https://www.rfc-editor.org/rfc/rfc9110.html#section-9.3.4
+     */
+    public function put(string $uri, string|array $params): void
+    {
+        $this->makeRequest("PUT", $uri, $params);
+    }
+
+    /**
+     * Make DELETE request
+     *
+     * @param string $uri
+     * @return void
+     *
+     * @url https://www.rfc-editor.org/rfc/rfc9110.html#section-9.3.5
+     */
+    public function delete($uri)
```

Comment:
Add `void` as the function return type 
```suggestion
    public function delete($uri): void
```

Reply:
As per the [documentation](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/DELETE), delete method might have Payload. So I think we can add `$params` as an optional variable. I think here @bgorski is just pointing out that we are not passing `$params` but using it in the makeRequest method.

Reply:
@engcom-Hotel `$params` will never be used in `makeRequest` for `DELETE` method. I think that it is not worth adding this parameter to the function description, because it will only confuse

Reply:
Hello @Den4ik, actually we are referring to this [documentation](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/DELETE) for `DELETE` method. And according to this, `DELETE` method might have Request. Thats I made this review comment.

Reply:
Hi @engcom-Hotel,

Thanks for your comment. Please compare the documentation for the DELETE request with the [documentation for the GET request](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/GET):

> An optional query component preceded by a question mark (?)

If you look at the GET request implementation, you will see that it doesn't contain any parameters.
The DELETE request implementation follows the same style as the GET request, and query parameters (preceded by a question mark) should be included in $uri.